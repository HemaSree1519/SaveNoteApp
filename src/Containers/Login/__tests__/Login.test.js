import React from 'react';
import {shallow} from 'enzyme';
import Index from '../index';

describe('Login', () => {
    it('should render correctly', () => {
        const component = shallow(<Index/>);
        expect(component).toMatchSnapshot();
    });
    it("should Login with given details", () => {
        const wrapper = shallow(<Index/>);
        const instance = wrapper.instance();
        instance.handleLogin = jest.fn(instance.handleLogin);
        wrapper.find('Input[name="email"]').simulate('change', {target: {value: 'Tester'}});
        wrapper.find('Input[name="password"]').simulate('change', {target: {value: 'password'}});
        wrapper.find('Button').simulate('click');
        expect(instance.handleLogin).toHaveBeenCalled();
    });
});