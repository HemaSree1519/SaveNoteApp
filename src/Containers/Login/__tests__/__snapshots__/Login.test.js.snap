exports[`Login should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        false,
        <Col
          tag="div"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }>
          <FormGroup
                    tag="div">
                    <Input
                              name="email"
                              onChange={[Function]}
                              placeholder="example@gmail.com"
                              type="email" />
          </FormGroup>
</Col>,
        <Col
          tag="div"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }>
          <FormGroup
                    tag="div">
                    <Input
                              name="password"
                              onChange={[Function]}
                              placeholder="Password"
                              type="password" />
          </FormGroup>
</Col>,
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button">
          Login
</Button>,
      ],
      "tag": "form",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FormGroup
            tag="div">
            <Input
                        name="email"
                        onChange={[Function]}
                        placeholder="example@gmail.com"
                        type="email" />
</FormGroup>,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              name="email"
              onChange={[Function]}
              placeholder="example@gmail.com"
              type="email" />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "example@gmail.com",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FormGroup
            tag="div">
            <Input
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password" />
</FormGroup>,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password" />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Login",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          false,
          <Col
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }>
            <FormGroup
                        tag="div">
                        <Input
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="example@gmail.com"
                                    type="email" />
            </FormGroup>
</Col>,
          <Col
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }>
            <FormGroup
                        tag="div">
                        <Input
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password" />
            </FormGroup>
</Col>,
          <Button
            color="secondary"
            onClick={[Function]}
            tag="button">
            Login
</Button>,
        ],
        "tag": "form",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FormGroup
              tag="div">
              <Input
                            name="email"
                            onChange={[Function]}
                            placeholder="example@gmail.com"
                            type="email" />
</FormGroup>,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                name="email"
                onChange={[Function]}
                placeholder="example@gmail.com"
                type="email" />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "example@gmail.com",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FormGroup
              tag="div">
              <Input
                            name="password"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password" />
</FormGroup>,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password" />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Login",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
